internals/cli/cmd_run.go

```diff
@@ -229,21 +229,39 @@ func runDaemon(rcmd *cmdRun, ch chan os.Signal, ready chan<- func()) (err error)
 
 	logger.Debugf("activation done in %v", time.Now().Truncate(time.Millisecond).Sub(t0))
 
+	// The "stop" channel is used by the "enter" command to stop the daemon.
+	var stop chan struct{}
+	if ready != nil {
+		stop = make(chan struct{}, 1)
+	}
+	notifyReady := func() {
+		ready <- func() { close(stop) }
+		close(ready)
+	}
+
 	if !rcmd.Hold {
+		// Start the default services.
 		servopts := client.ServiceOptions{}
 		changeID, err := rcmd.client.AutoStart(&servopts)
 		if err != nil {
 			logger.Noticef("Cannot start default services: %v", err)
-		} else {
+		} else if ready == nil {
 			logger.Noticef("Started default services with change %s.", changeID)
+		} else {
+			// If ready != nil (case for "enter" command), wait for the default
+			// services to start and then notify on ready channel.
+			go func() {
+				_, err := rcmd.client.WaitChange(changeID, nil)
+				if err != nil {
+					logger.Noticef("Cannot wait for autostart change %s", changeID)
```

Comment:
This is reflected in the above suggestion, but we should include the `err` value here:
```suggestion
					logger.Noticef("Cannot wait for autostart change %s: %v", changeID, err)
```